File: heritage_simple/java/redefinition/Animal.java
public class Animal {
    protected String nom;
    protected int nombreDePattes;

    public Animal(){
        nom="defaut";
        nombreDePattes=0;
    }

    public Animal(String nom, int nombreDePattes){
        this.nom=nom;
        this.nombreDePattes=nombreDePattes;
    }

    public Animal(Animal animal){
        nom=animal.nom;
        nombreDePattes=animal.nombreDePattes;
    }

    public String getNom(){
        return nom;
    }

    public int getNombreDePattes(){
        return nombreDePattes;
    }

    public void setNom(String nom){
        this.nom=nom;
    }

    public void setNombreDePattes(int nombreDePattes){
        this.nombreDePattes=nombreDePattes;
    }

    //méthode que l'on va redéfinir
    public void crier(){
        System.out.println("Cri d'un animal");
    }
}

File: heritage_simple/cpp/redefinition/Animal.cpp
#include "Animal.h"

Animal::Animal():nom("defaut"),nombreDePatte(0){};
Animal::Animal(string nom, int nombreDePatte):nom(nom),nombreDePatte(nombreDePatte){};
Animal::Animal(Animal const& animal):nom(animal.nom),nombreDePatte(animal.nombreDePatte){};

string Animal::getNom() const{return nom;}
int Animal::getNombreDePatte() const{return nombreDePatte;}

void Animal::setNom(string nom){this->nom=nom;}
void Animal::setNbPatte(int nombreDePatte){this->nombreDePatte=nombreDePatte;}

void Animal::crier() const{
    cout<<"Reeuh"<<endl;
}

Animal::~Animal(){};